src/v/cluster/client_quota_backend.cc

```diff
@@ -0,0 +1,54 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "cluster/client_quota_backend.h"
+
+#include "cluster/client_quota_serde.h"
+#include "cluster/commands.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/util/variant_utils.hh>
+
+#include <optional>
+
+namespace cluster::client_quota {
+
+ss::future<std::error_code> backend::apply_update(model::record_batch batch) {
+    auto cmd = co_await cluster::deserialize(std::move(batch), commands);
+    co_await _quotas.invoke_on_all([cmd{std::move(cmd)}](store& quota_store) {
+        return ss::visit(
+          std::move(cmd), [&quota_store](alter_quotas_delta_cmd cmd) {
```

Comment:
This move doesn't move, you have a copy.

Reply:
Done. Removed the `std::move` so `ss::visit` will now take `cmd` by `const&`.

Reply:
> Done. Removed the std::move so ss::visit will now take cmd by const&.
It still accepts by value though, so you still get the copy.

Reply:
Right, because of the inner lambda taking `cmd` by value. I've fixed this now.